packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -138,6 +140,58 @@ export class Rule extends Resource implements IRule {
 
     const roleArn = targetProps.role ? targetProps.role.roleArn : undefined;
 
+    if (targetProps.target) {
+      const targetStack = Stack.of(targetProps.target);
+      const sourceStack = Stack.of(this);
+      const sourceAccount = sourceStack.account;
+      if (targetStack.account !== sourceAccount) {
+        // cross-account event - strap in, this works differently than regular events!
+        // based on:
+        // https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/CloudWatchEvents-CrossAccountEventDelivery.html
+
+        // the _actual_ target is just the event bus of the target's account
+        this.targets.push({
+          id,
+          arn: `arn:aws:events:${targetStack.region}:${targetStack.account}:event-bus/default`,
+        });
+
+        // Grant the source account permissions to the event bus of the target account.
+        // Do it in a separate stack, as it needs to be created before the rule containing the above event bus target
+        // (CloudWatch verifies whether you have permissions to the target on rule creation),
+        // but it's very likely the target stack has a dependency on the source (= this) stack
+        // (that's the case with CodePipeline, for example)
+        const stackId = `EventBusPolicy-${sourceAccount}`;
+        let eventBusPolicyStack: Stack = this.node.tryFindChild(stackId) as Stack;
+        if (!eventBusPolicyStack) {
+          eventBusPolicyStack = new Stack(this, stackId, {
+            env: {
+              account: targetStack.account,
+              region: targetStack.region,
+            },
+            stackName: `${targetStack.stackName}-EventBusPolicy-support-${sourceAccount}`,
```

Comment:
I'd love  for this to be a nested stack of `targetStack`, so we  don't  have to invent a name. 

...but nested stacks may not  work  very well yet, so you're going to have to do some implementation work.

Reply:
Discussed offline - this can't be a nested stack of `targetStack`, as it has to be deployed before `sourceStack`, and it's common that `targetStack` depends on `sourceStack`.

Let me know if the comment above describes that clearly enough, or does it need some re-wording.

Reply:
Related: #2821

Reply:
I think  your GitHub comment is  much clearer than the code comment :)

Reply:
Reworded the comment. Let me know if it's OK now.